projects/fsastorefrontlib/src/cms-components/my-account/policy/policy-details/policy-details.component.ts

```diff
@@ -25,30 +27,39 @@ export class PolicyDetailsComponent implements OnInit, OnDestroy {
     protected config: OccConfig,
     protected changeRequestService: ChangeRequestService,
     protected translationService: FSTranslationService,
-    protected fileService: FileService
+    protected fileService: FileService,
+    protected activatedRoute: ActivatedRoute,
+    private winRef: WindowRef
   ) {}
 
   policy$;
   subscription = new Subscription();
   baseUrl: string;
+  originPolicyPage = this.winRef.sessionStorage.getItem(ORIGIN_POLICIES_PAGE);
 
   ngOnInit(): void {
+    if (!this.originPolicyPage) {
```

Comment:
Using information about origin of users visit. If he/she came from My policies page, they won't be automatically redirected to Change Coverage page. If they came via direct link, there won't be anything in session storage, and the user will be redirected.